assets/src/globals.js

```diff
@@ -0,0 +1,28 @@
+import { createMuiTheme } from '@material-ui/core/styles'
+import defaultPalette from './defaultPalette'
+
+const mylaGlobals = JSON.parse(document.getElementById('myla_globals').textContent)
+
+/*
+Frozen to prevent unintentional changes to this object. This object is strictly readonly.
+mylaGlobals should ONLY be accessed in globals.js, and nowhere else.
+*/
+const user = Object.freeze({
+  username: mylaGlobals.username,
+  admin: mylaGlobals.is_superuser,
+  relatedCourses: mylaGlobals.user_courses_info,
+  isSuperuser: mylaGlobals.is_superuser,
+  isLoggedIn: !!mylaGlobals.username,
+  helpURL: mylaGlobals.help_url
+})
+
+const palette = JSON.parse(JSON.stringify(defaultPalette))
```

Comment:
@justin0022, I know I probably isn't critical to clone the object here (since we don't really need a stable `defaultPalette` for any reason I can think of), but it seemed like a best practice. Do you think it's fine to use `JSON.stringify` here? I also looked at `_.cloneDeep`, but it seemed somewhat heavyweight for this purpose. We don't list `lodash` as a dependency in `package.json` (though it's available, I believe because it's a dependency of many of the packages we use) and it's not used anywhere else.

Reply:
Hmm I'm not sure if this is necessary, maybe you could simply freeze it?

Reply:
How would you approach that? I couldn't freeze `defaultPalette`, since then I couldn't update that object (stored in a new variable) with a specified primary color. I'm also not sure I can freeze `palette`, since I'm guessing `createMuiTheme` does some magic to fill in things you didn't specify.

I think the question is, should I bother cloning at all? It's not much code and leaves open some more possibilities, so I guess I leaned in that direction.

Reply:
Ya that's fair, I think you can just leave it as is for now.